rtl/eclrtl/rtlfield.cpp

```diff
@@ -298,14 +298,12 @@ size32_t RtlStringTypeInfo::build(ARowBuilder &builder, size32_t offset, const R
         builder.ensureCapacity(offset+size+sizeof(size32_t), field->name->str());
         byte *dest = builder.getSelf()+offset;
         rtlWriteInt4(dest, size);
-#if 0
-        // NOTE - you might argue that we should convert the incoming data to EBCDIC. But it seems more useful to
-        // define the semantics as being that the IFieldSource should return EBCDIC if you have declared the matching field as EBCDIC
-        // (otherwise, why did you bother?)
+        // NOTE - it has been the subject of debate whether we should convert the incoming data to EBCDIC, or expect the IFieldSource to have already returned ebcdic
```

Comment:
The way I see it the RtlStringTypeInfo structures are describing how ECL will store the information, making no judgement on how the source has stored it.

We have two choices I think:

1. Expand the IFieldSource interface so that when we are going to store an EBCDIC string we call source.getEBCDICresult rather than getStringResult
2. Take this change.

The language you are calling very likely has no concept of EBCDIC or big-endian integers - chances are plugin implementers will just do the conversion there and negate any possible benefit of option 1 (or some would, some wouldn't). ECL code, OTOH, can easily be written to do the conversion if that is what is needed.

I'm pretty convinced that the change that was made before was wrong, and undoing it in this PR is correct. I doubt anyone is really using this feature (deliberately) anyway.